core/Plugin/ControllerAdmin.php

```diff
@@ -170,6 +171,25 @@ private static function notifyIfURLIsNotSecure()
         Notification\Manager::notify('ControllerAdmin_HttpIsUsed', $notification);
     }
 
+    private static function notifyIfDevelopmentModeOnButNotInstalledThroughGit()
+    {
+        if (!Piwik::hasUserSuperUserAccess()) {
+            return;
+        }
+
+        if (SettingsPiwik::isGitDeployment()) {
+            return;
+        }
```

Comment:
I haven't tested locally, so I might be missing something, but shouldn't this also check `if (Development::isEnabled())`?

Reply:
you were right @diosmosis, I added the extra validation